src/updater/updater.cpp

```diff
@@ -145,13 +145,22 @@ void checkWeightsConsistency(
 
 static const constexpr std::size_t LUA_SOURCE = 0;
 
+void recordsUpdatedNodes(NodeSetPtr node_updated, const NodeID n)
```

Comment:
Frequently copy shared_ptr costs much time, too heavy. Better to use `NodeSetPtr& node_updated`  instead.

Reply:
Only concerns performance here. I'm thinking that it might be better to use `std::unique_ptr` for `NodeSet` instead of `shared_ptr`, then pass in the `unique_ptr` by reference always.

Reply:
Good topic for pure technical discuss.

I just finished to read part of [c++ core guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines), one of the item I pay attention to is: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-sharedptrparam

I have a think about following code's differences
```
void share(std::shared_ptr<Widget> shaWid);   // 1
void reseat(std::shared_ptr<Widget>& shaWid);   // 2
void mayShare(const std::shared_ptr<Widget>& shaWid);   // 3
```
I think while using std::shared_ptr<Widget>&, the idea I want to express is, inside function I might reset the shared pointer.  The inside function is not shared owner of the widget because it will not change the reference counter.  The inside function will not guarantee that widget object is still alive during running time.  That's why I don't think passing by reference is clear here.